src/controller/CommissioningWindowOpener.h

```diff
@@ -104,8 +106,8 @@ class CommissioningWindowOpener
      */
     CHIP_ERROR OpenCommissioningWindow(NodeId deviceId, System::Clock::Seconds16 timeout, uint32_t iteration,
                                        uint16_t discriminator, Optional<uint32_t> setupPIN, Optional<ByteSpan> salt,
-                                       Callback::Callback<OnOpenCommissioningWindow> * callback, SetupPayload & payload,
-                                       bool readVIDPIDAttributes = false);
+                                       Optional<ByteSpan> verifier, Callback::Callback<OnOpenCommissioningWindow> * callback,
```

Comment:
we had 9 parameters to a method. Now we have 10 ... why is this ok to keep going this route? This is not easy to read and maintain. For example

```
 mWindowOpener->OpenCommissioningWindow(mNodeId, System::Clock::Seconds16(mCommissioningWindowTimeout), mIteration,
                                                      mDiscriminator, NullOptional, NullOptional, NullOptional,
                                                      &mOnOpenCommissioningWindowCallback, ignored,
                                                      /* readVIDPIDAttributes */ true);
```

there `NullOptional` is repeated 3 times without any indication of what they mean. Let's start making code cleaner: please create a `params` structure that uses a builder pattern and then you will not need to change every single call site.

Even more, if we are willing to break ABI compatibility (we seem to want to since we are changing this call anyway) we may as well make the params builder the only call that someone can make.

Reply:
Added a params class and brought back the stable API for external consumers in https://github.com/project-chip/connectedhomeip/pull/34209/commits/5d4ce206b24d3a3a4201e6d1cdc816ca6e7f8b15